src/app/shared/user-details.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject } from "rxjs";
+
+@Injectable({
+  providedIn: "root",
+})
+export class UserDetailsService {
+  users = new BehaviorSubject([
```

Comment:
Make `users` private and define a function that returns the value of the users. This function is what other components should make use in order to get and display data.
In the same way if you are updating this behaviour subject, you should define a function specifically for updating it and other components should make use of the function to update `users`.

Reply:
This change has been made.